spec/models/sight_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe Sight, type: :model do
+
+  it 'animal is valid with all attr filled' do
+    animal = Animal.create name: 'lion', latin: 'latin', kingdom: 'king'
+    expect(animal).to be_valid
+  end
+
+  it 'animal is valid when all attr NOT filled' do
+    animal = Animal.create name: 'tiger', latin: 'tiger latin'
+    expect(animal.errors[:kingdom]).to_not be_empty
+  end
+
+  it 'sight is valid with all attr filled' do
+    animal = Animal.create name: 'lion', latin: 'latin', kingdom: 'king'
+    sight = Sight.create animal_id: animal.id, date: Date.today, latitude: 5, longitude: 5
+    expect(sight).to be_valid
+    
+  end
+
+  it 'sight is should not be valid when all attr are NOT filled' do
+    animal = Animal.create name: 'lion', latin: 'latin', kingdom: 'king'
+    sight = Sight.create animal_id: animal.id, date: Date.today, longitude: 5
+    expect(sight.errors[:latitude]).to_not be_empty
+
+  end
+
+  it 'latin and name cannot be same' do
+    animal = Animal.create name: 'lion', latin: 'lion', kingdom: 'king'
+    expect(animal.errors[:name]).to_not be_empty
+    expect(animal.errors[:latin]).to_not be_empty
+
+  end
+
+  it 'latin and name must be unique' do
+    animal1 = Animal.create name: 'lion', latin: 'lion', kingdom: 'king'
+    animal2 = Animal.create name: 'lion', latin: 'lion', kingdom: 'king'
+    expect(animal1).not_to be_valid
+    expect(animal2).not_to be_valid
```

Comment:
Awesome job with all your testing Sami! I also tested the Validations and they all seemed to work great!! The only thing that I am not sure about is about the expect on line 39 and 40. Maybe you would expect the first variable (animal1) to be valid but the second one to not be. So maybe something along the lines of:
expect(animal2.erros[:name].to_not be_empty
expect(animal2.erros[:lastin].to_not be_empty
Like I said I'm not 100% sure on that one but it just seems off that animal1 in your test would not be valid when you need it to be valid and the second one to be causing the error. Great job though! All the validations work and the testing looks great!

Reply:
awesome job including tests! Great start. I would suggest making assertions on error messages since you're passing them into your validation errors